### PR TITLE
Rename DoSync to Seq and add context + timeouts

### PR DESCRIPTION
**Pull request by mkcp**
This PR moves do-sync to a file and type called Seq, introduces SeqConfig, allows for Seq to be constructed with a context, and enables the entire block to be timed out at the Seq level interrupting all runners.

### CODE REVIEW

File: runner/do/seq.go [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+			// Add an op for this failed DoSync at the end of the slice
+			results[o.Identifier] = o
+			return op.New(s.ID(), results, op.Fail, ChildRunnerError{
+				Parent: s.ID(),
+				Child:  o.Identifier,
+				err:    o.Error,
+			}, runner.Params(s), time.Time{}, time.Now())
+		}
+	}
+	// Return runner ops, adding one at the end for our successful DoSync run
```
Comment (nwchandler) [inline runner/do/seq.go:116]: Nit: The comment still references DoSync

File: runner/do/seq.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+func (s Seq) run() op.Op {
+	results := make(map[string]any, 0)
+	for _, r := range s.Runners {
+		s.log.Info("running operation", "runner", r.ID())
+		o := r.Run()
+		// If any result op is not Success, abort and return all existing ops
+		if o.Status != op.Success {
+			// Add an op for this failed DoSync at the end of the slice
```
Comment (nwchandler) [inline runner/do/seq.go:107]: Nit: The comment still references DoSync

File: hcl/hcl.go [hunk 3]
```diff
@@ -104,13 +103,13 @@ type Do struct {
 	// Redactions   []Redact      `hcl:"redact,block" json:"redactions,omitempty"`
 }
 
-type DoSync struct {
+type Seq struct {
```
Comment (nwchandler) [inline hcl/hcl.go:106]: I didn't see a Timeout field on this - should there be one?

**Review by nwchandler — COMMENTED** [review]
I left a few comments inline. I also did a quick grep on the branch and saw one other reference to DoSync in a file that wasn't changed: agent/manifest.go at line 22.

**Comment by mkcp** [comment]
PR up that addresses feedback!

**Review by nwchandler — APPROVED** [review]
👍

**Pull request merged**